LegiScan API documentation: https://legiscan.com/documentation/legiscan

In [25]:
import os
import time
import requests
import pandas as pd
from datetime import date
from dotenv import load_dotenv

load_dotenv() 
print ("Libraries imported")

Libraries imported


In [26]:
API_KEY = os.getenv("LEGI_API_KEY")

In [27]:
url = "https://api.legiscan.com/"
## I need to get a access code from the getDatasetList url used to call the GetDataset url.

In [28]:
params = {
    "key": API_KEY,
    "op": "getDatasetList",
    "state": "US"
}

response = requests.get(url, params=params)

print(response.status_code)

data = response.json()
print(data.keys())


200
dict_keys(['status', 'datasetlist'])


In [29]:
status = data["status"]
print(status)

OK


In [30]:
print(data["datasetlist"][0])


{'state_id': 52, 'session_id': 2199, 'year_start': 2025, 'year_end': 2026, 'prefile': 0, 'sine_die': 0, 'prior': 0, 'special': 0, 'session_tag': 'Regular Session', 'session_title': '2025-2026 Regular Session', 'session_name': '119th Congress', 'dataset_date': '2026-09-06', 'dataset_hash': '99dcea213d48715d5dc603e1a48da9ac', 'dataset_size': 62930520, 'dataset_size_csv': 4403830, 'access_key': '7GNHYv7EcN1RPXURARflAn'}


9/11/2026: <br>
    session_id: 2199 <br>
    access_key: '7GNHYv7EcN1RPXURARflAn' <br>
    dataset_hash: '99dcea213d48715d5dc603e1a48da9ac' <br>

In [48]:
print(data["datasetlist"][0]["dataset_hash"])

99dcea213d48715d5dc603e1a48da9ac


In [31]:
key = data["datasetlist"][0]["access_key"]
print(key)

7GNHYv7EcN1RPXURARflAn


In [33]:
session_id = data["datasetlist"][0]["session_id"]
print(session_id)

2199


In [35]:
params2= {
    "key": API_KEY,
    "op": "getDatasetRaw",
    "id": session_id,
    "access_key": key
}

response2 = requests.get(url, params=params2)

200
dict_keys(['status', 'datasetlist'])


In [43]:
print(response2.status_code)

data2 = len(response2.content)
print(data2)

200
62930520


In [46]:
response2.headers["Content-Type"]

'application/zip'

In [50]:
dataset_date = data["datasetlist"][0]["dataset_date"]

In [51]:
open(f"legiscan_data/getDatasetRaw{dataset_date}.zip", "wb").write(response2.content)
print(f"File saved to legiscan_data/getDatasetRaw{dataset_date}.zip")

File saved to legiscan_data/getDatasetRaw2026-09-06.zip
